File: 1470.rb
# @param {Integer[]} nums
# @param {Integer} n
# @return {Integer[]}
def shuffle(nums, n)
#   result = []
#   (nums.length / 2).times { |i| result << nums[i] << nums[i + n] }
#   result
  nums[...n].zip(nums[n..]).flatten
end

# p shuffle([2,5,1,3,4,7], 3)
# p shuffle([1,2,3,4,4,3,2,1], 4)
# p shuffle([1,1,2,2], 2)

File: 1470.go
func shuffle(nums []int, n int) []int {
  result := make([]int, 0, n * 2)
  for i, j := 0, n; i < n; i, j = i + 1, j + 1 {
    result = append(result, nums[i], nums[j])
  }
  return result
}
